relay-system/src/controller.rs

```diff
@@ -103,20 +104,27 @@ impl Controller {
 
     /// Starts an actix system and runs the `factory` to start actors.
     ///
+    /// The function accepts the reference to [`tokio::runtime::Handle`] from the new tokio 1.x
+    /// runtime, which we enter before the factory is run, to make sure that to systems, old and
+    /// new one is available.
+    ///
     /// The factory may be used to start actors in the actix system before it runs. If the factory
     /// returns an error, the actix system is not started and instead an error returned. Otherwise,
     /// the system blocks the current thread until a shutdown signal is sent to the server and all
     /// actors have completed a graceful shutdown.
-    pub fn run<F, R, E>(factory: F) -> Result<(), E>
+    pub fn run<F, R, E>(
+        handle: &tokio::runtime::Handle,
+        sys: SystemRunner,
```

Comment:
Why not create the runtime and the `actix::SystemRunner` inside `run()`? Is there a use case where `run()` recycles an existing runtime?

Reply:
This is because, if I remember it correctly - I use in `relay-server/src/lib.rs` following code
```rust
let runtime = utils::create_runtime("http-server-handler", 1);
```
And it requires that old system is already available at this point. And also I wanted to clean a little bit controller, and make sure that controller knows less about the system it runs on, just accept the system as the parameter.

Reply:
But isn't spinning up the actix system kind of what the `Controller` is for? Looking at its docs

https://github.com/getsentry/relay/blob/dd9f998ee324a710528df2fa76f87a64614a45ef/relay-system/src/controller.rs#L47

Reply:
I've changed this doc and added more docs to clarify why it's done this way now.